packages/atclient/include/atclient/atclient.h

```diff
@@ -114,10 +114,12 @@ int atclient_get_selfkey(const atclient *atclient, atclient_atkey *atkey, char *
  * @param value the buffer to hold value gotten from atServer
  * @param valuelen the buffer length allocated for the value
  * @param valueolen the output length of the value gotten from atServer
+ * @param bypasscache true if you want to bypass the cached publickey, that might be on your atServer, and get the most
+ * up-to-date value straight from the atServer that the publickey sits on, false otherwise
  * @return int 0 on success
  */
 int atclient_get_publickey(const atclient *atclient, const atclient_atkey *atkey, char *value, const size_t valuelen,
```

Comment:
We have to think about how to join the two functions, because I had already created a function for this in https://github.com/atsign-foundation/at_c/pull/111 (although it did not use `const atclient_atkey *atkey` as input)

https://github.com/atsign-foundation/at_c/blob/4a8487b31378805c666dacbc5003359fb2356404/packages/atclient/src/atclient.c#L594

Reply:
This is for public atkey not public encryption key?

Reply:
This is actually more a suggestion. It is for public encryption key, but isnt it a public key in some way? I think it can be converted into the same by changing a few lines (maybe we could get pub_enc_key using get_pub_key, reducing the amount of code in atclient)

Reply:
Wouldn't the simple solution be to use `atclient_get_publickey` inside of `atclient_get_public_encryption_key`, since `_get_public_encryption_key` is a subset of `_get_publickey` ?

Reply:
I think so too

Reply:
Once this PR is merged, maybe we could write a ticket on refactoring `_get_public_encryption_key` to use `_get_publickey`, and then mark this conversation as resolved?

Reply:
> Once this PR is merged, maybe we could write a ticket on refactoring `_get_public_encryption_key` to use `_get_publickey`, and then mark this conversation as resolved? @realvarx

That would be fine